### PR TITLE
Revamp `Current Channel` page

### PR DESCRIPTION
**Pull request by wedebe**
Revamp `Current Channel` page

This change:
- reorganises informational items into `Channel`, `Tuner` and `Satellite` categories
- replaces `True`, `False` and `0` values with `Yes` or `No`
- shows hex values for satellite information
- provides more specific `audio` information (`track` and `language`)
- updates related translation strings (untranslated)

This change does not (yet) completely the refactor the page

Updated:

Previous:

### CODE REVIEW

**Comment by jbleyel** [comment]
Please don’t change the po files and the pot file at once. This is a problem of weblate. You need to change the pot only, then wait until weblate has changed the po files and then push your changed po files.

**Comment by jbleyel** [comment]
Please check the pot file. There is a duplicate string.

**Comment by jbleyel** [comment]
/home/weblate/data/vcs/e2openplugin-openwebif/e2openplugin-openwebif/locale/OpenWebif.pot:2608: duplicate message definition...
/home/weblate/data/vcs/e2openplugin-openwebif/e2openplugin-openwebif/locale/OpenWebif.pot:1293: ...this is the location of the first definition
msgmerge: found 1 fatal error

File: locale/OpenWebif.pot [hunk 1]
```diff
@@ -2599,6 +2605,9 @@ msgstr ""
 msgid "minute"
 msgstr ""
 
+msgid "No"
```
Comment (jbleyel) [inline locale/OpenWebif.pot:2608]: Hi @wedebe , this is the duplicate.

**Comment by jbleyel** [comment]
Fixed !!

PO Files also updated from weblate.

**Comment by tinola** [comment]
> Revamp `Current Channel` page
> 
> This change:
> 
>     * reorganises informational items into `Channel`, `Tuner` and `Satellite` categories
> 
>     * replaces `True`, `False` and `0` values with `Yes` or `No`
> 
>     * shows hex values for satellite information
> 
>     * provides more specific `audio` information (`track` and `language`)
> 
>     * updates related translation strings (untranslated)
> 
> 
> This change does not (yet) completely the refactor the page
> 
> Updated: 
> 
> Previous: 

Hi @wedebe :)

Reorganization in current.tmpl file does not work on system with Python 3.x. Spinner "loading" shows infinitely.
Debug says:

`Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/twisted/protocols/basic.py", line 548, in dataReceived
    why = self.lineReceived(line)
  File "/usr/lib/python3.10/site-packages/twisted/web/http.py", line 2243, in lineReceived
    self.allContentReceived()
  File "/usr/lib/python3.10/site-packages/twisted/web/http.py", line 2369, in allContentReceived
    req.requestReceived(command, path, version)
  File "/usr/lib/python3.10/site-packages/twisted/web/http.py", line 1003, in requestReceived
    self.process()
--- <exception caught here> ---
  File "/usr/lib/python3.10/site-packages/twisted/web/server.py", line 229, in process
    self.render(resrc)
  File "/usr/lib/python3.10/site-packages/twisted/web/server.py", line 294, in render
    body = resrc.render(self)
  File "/usr/lib/enigma2/python/Plugins/Extensions/OpenWebif/controllers/base.py", line 243, in render
    out = self.loadTemplate(module, self.path, data)
  File "/usr/lib/enigma2/python/Plugins/Extensions/OpenWebif/controllers/base.py", line 132, in loadTemplate
    return str(Template(file=vp, searchList=[args]))
  File "/usr/lib/python3.10/site-packages/Cheetah/Template.py", line 1053, in __unicode__
    return getattr(self, mainMethName)()
  File "_usr_lib_enigma2_python_Plugins_Extensions_OpenWebif_controllers_views_responsive_ajax_current_tmpl.py", line 429, in respond

builtins.TypeError: %x format: an integer is required, not str

`
Browser debug says error 500 (Internal Server Error).

**Comment by jbleyel** [comment]
No issue here.

Bildschirmfoto 2022-04-08 um 15 10 04

**Comment by tinola** [comment]
Hmm, interesting...
What version of Python do you have on board? I've got 3.10. May be there is a key?... 🧐

**Comment by jbleyel** [comment]
Python 3.10.2

**Comment by tinola** [comment]
Could you please provide your python-twisted version as well?...

**Comment by jbleyel** [comment]
That’s not a twisted issue.
There is an issue in the compiled template.

**Comment by tinola** [comment]
Yes, that's indeed not a twisted issue. The problem is still exists on Python 3.10.4... :/

**Comment by jbleyel** [comment]
let me check this again after update to 3.10.4

**Comment by wedebe** [comment]
@tinola, does the issue happen for every event on every channel?

**Comment by wedebe** [comment]
There may be an "N/A" or "0" value returned for one of the pid values. Thought I'd caught them all but seems there may be one I've overlooked.

**Comment by tinola** [comment]
Woah! It's very very strange... Indeed it not happen on every channel! But what I observed at quick look - channels on HotBird 13.0E are failing, when I trying eg. arteHD or 3sat HD on Astra 19.2E there's no problem!...
Please try (if you can) for example FTA channel 4Fun.TV or STARS.TV from HotBird 13.0E - on those current page fails.

**Comment by wedebe** [comment]
I'm not sure what the API can return for apid/vpid/onid etc, could someone provide owif's url if it provides one, please?

**Comment by tinola** [comment]
Here you are, channel Discovery Science HD, HB 13.0E:

```
{
 "ppid": "31",
 "vpid": "6d",
 "apid": "d1",
 "host": "vu-uno4k-se"
}
```

**Comment by wedebe** [comment]
> Here you are, channel Discovery Science HD, HB 13.0E:
> 
> ```
> {
>  "ppid": "31",
>  "vpid": "6d",
>  "apid": "d1",
>  "host": "vu-uno4k-se"
> }
> ```

Great, thanks. I know what the problem is.

**Comment by tinola** [comment]
Well, how can we detect when those values come in hex or dec?...

Yeah, I truncated the `host` string, there is FQDN of course. Hostname is provided by my DNS server.

**Comment by wedebe** [comment]
> Well, how can we detect when those values come in hex or dec?...
> 
> Yeah, I truncated the `host` string, there is FQDN of course. Hostname is provided by my DNS server.

My box returns `pid` values without quotes, hex values provided above are quoted, but I'm not sure if any images return unquoted `0x...` values.

**Comment by tinola** [comment]
Hmm, I don't thing it's related to values in quotes...

Working example (`/api/getpid`):

```
{
 "ppid": "1964",
 "vpid": "196e",
 "apid": "1978",
 "host": "vu-uno4k-se.mydomain.tld"
}
```

Current page displays:

obraz

Not working example (`/api/getpid`):

```
{
 "ppid": "31",
 "vpid": "6d",
 "apid": "d1",
 "host": "vu-uno4k-se.mydomain.tld"
}
```

Very very interesting.

**Comment by jbleyel** [comment]
Strange:

```
>>> "%04x [%d]" % (49,49)
'0031 [49]'
>>> "%04x [%d]" % (109,109)
'006d [109]'
>>> "%04x [%d]" % (209,209)
'00d1 [209]'
```

**Comment by wedebe** [comment]
> Working example (`/api/getpid`):
> 
> ```
>  "vpid": "196e",

I think this `e` might be considered as an exponent in this case... So where it seems to _work_, the converted value displayed  is incorrect 😬

I think I'll disable the conversion and just show what the image tells OWIF until I can investigate each image's handling of this.

**Pull request merged**